lib/LedgerSMB/Scripts/payment.pm

```diff
@@ -1216,10 +1219,18 @@ for (my $i=1 ; $i <= $request->{overpayment_qty}; $i++) {
      # Now we split the account selected options, using the namespace the if statement
      # provides for us.
      $request->{"overpayment_topay_$i"} = LedgerSMB::PGNumber->from_input($request->{"overpayment_topay_$i"});
-     $request->{"overpayment_account_$i"} =~ /^(\d+)--*/;
-     my $id = $1;
-     $request->{"overpayment_cash_account_$i"} =~ /^(\d+)--*/;
-     my $cashid = $1;
+     my $id;
+     if ( $request->{"overpayment_account_$i"} =~ /^(\d+)--*/) {
+        $id = $1;
+     } else {
+        die "Error: Invalid data";
+     }
+     my $cashid;
+     if ( $request->{"overpayment_cash_account_$i"} =~ /^(\d+)--*/) {
+        $cashid = $1;
+     } else {
+        die "Error: Invalid data";
```

Comment:
Will change.

If I don't set up one or more G/L overpayment accounts, where do I apply the excess of a payment?   A question for lsmb.user or .dev.